core/libp2p/peer/peer_info.hpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PEER_INFO_HPP
+#define KAGOME_PEER_INFO_HPP
+
+#include <set>
+#include <unordered_set>
+#include <vector>
+
+#include <gsl/span>
+#include <outcome/outcome.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/multi/multihash.hpp"
+
+namespace libp2p::peer {
+  /**
+   * Information about the peer in the network
+   */
+  class PeerInfo {
+   private:
+    using PeerId = multi::Multihash;
```

Comment:
it is bad Idea IMO. 

maybe use composition? just make single field class - multihash. If all PeerIds require sha256 hash, then this logic can be present in this class, instead class-clients

Reply:
A good suggestion, going to make it